legend-engine-xt-serviceStore-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/extensions/IServiceStoreGrammarParserExtension.java

```diff
@@ -30,8 +47,48 @@ static List<IServiceStoreGrammarParserExtension> getExtensions()
         return Lists.mutable.withAll(ServiceLoader.load(IServiceStoreGrammarParserExtension.class));
     }
 
-    default List<Function<String, SecurityScheme>> getExtraSecuritySchemesParsers()
+    static SecurityScheme process(SecuritySchemeSourceCode code, List<Function<SecuritySchemeSourceCode, SecurityScheme>> processors)
+    {
+        return process(code, processors, "Security Scheme");
+    }
+
+    default List<Function<SecuritySchemeSourceCode, SecurityScheme>> getExtraSecuritySchemesParsers()
+    {
+        return Collections.emptyList();
+    }
+
+    default List<Function2<String, Map<String,SecurityScheme>, SecuritySchemeRequirement>> getExtraSecurityParsers()
     {
         return Collections.emptyList();
     }
+
+    @Deprecated
+    default List<Function<SpecificationSourceCode, AuthenticationSpecification>> getExtraAuthenticationParsers()
+    {
+        return Collections.emptyList();
+    }
+
+    static <T extends SpecificationSourceCode, U> U process(T code, List<Function<T, U>> processors, String type)
```

Comment:
Is this used at multiple places, if not can we just have the body there and not complicate with generics and complicated code